model/dns_zone/dns_zone.rb

```diff
@@ -52,6 +52,40 @@ def delete_record(record_name:, type: nil, data: nil)
     incr_refresh_dns_servers
   end
 
+  def purge
```

Comment:
Maybe rename this to `purge_obsoleted_records`.

Reply:
Renamed this and the state in the nexus